cub/cub/block/block_load_to_shared.cuh

```diff
@@ -0,0 +1,228 @@
+// SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+
+#include <cub/config.cuh>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cub/util_ptx.cuh>
+#include <cub/util_type.cuh>
+
+// TODO Should I really include the whole cuda/memcpy_async instead? Or just copy the inline PTX?
+#include <cuda/__memcpy_async/cp_async_shared_global.h>
+#include <cuda/cmath>
+#include <cuda/memory>
+#include <cuda/ptx>
+#include <cuda/std/__algorithm/max.h>
+#include <cuda/std/bit>
+#include <cuda/std/cstdint>
+#include <cuda/std/span>
+
+#include <nv/target>
+
+CUB_NAMESPACE_BEGIN
+
+template <int BLOCK_DIM_X, int BLOCK_DIM_Y = 1, int BLOCK_DIM_Z = 1>
+struct BlockLoadToShared
+{
+private:
+  /// Constants
+  static constexpr int BLOCK_THREADS = BLOCK_DIM_X * BLOCK_DIM_Y * BLOCK_DIM_Z;
+  // TODO Should this be public?
+  static constexpr unsigned BLK_MINIMUM_ALIGN = 16u;
```

Comment:
No

Reply:
This is just a standin, not planning to leave it hardcoded 😂